### PR TITLE
add yarn lock file

### PR DESCRIPTION
**Pull request by erwinmombay**
see https://yarnpkg.com/ https://github.com/yarnpkg/yarn

yarn installs much faster than npm as well as having deterministic installs because of the additional information it stores in the yarn lock file compared to the shrinkwrap format npm uses.

here is an article for a cheatsheet of commands https://shift.infinite.red/npm-vs-yarn-cheat-sheet-8755b092e5cc#.7yzc63q1q

TLDR:
`npm install` -> `yarn`
`npm install ${npm-package} --save` -> `yarn add ${npm-package}`
`npm install ${npm-package} --save-dev` -> `yarn add ${npm-package} --dev`

even if you don't want to use yarn for development process, we're investigating using it on travis for a more deterministic build

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +52,6 @@
     "gulp-git": "1.7.2",
     "gulp-help": "1.6.1",
     "gulp-if": "2.0.0",
-    "gulp-image-diff": "0.3.0",
```
Comment (dvoytenko) [inline package.json]: Go ahead.

**Comment by cramforce** [comment]
Please describe in PR description how this would be expected to be used in our workflow.

**Comment by erwinmombay** [comment]
@cramforce done

**Comment by cramforce** [comment]
How does the lock file get updated?

**Comment by erwinmombay** [comment]
@cramforce haven't looked too much into it, but currently i know invoking `yarn add/remove` and `yarn` updates the yarn.lock file automatically

**Pull request closed without merging**